### ISSUE TITLE
Project level

### ISSUE THREAD

**Issue by karintang** [nus-cs2030/2021-s2#138]
In level 3, we need to implement a greedy customer. However, we are not allowed to use instanceof in our code. Are we allowed to use other ways to check if the customer is a greedy customer or not?

**Comment by gableejh** [comment]
Yes you'll have to use other ways since instanceof is not allowed. For me, I created a method in GreedyCustomer and Customer which returns a String. I then use Customer.method_name().equals("Greedy") to check if it's a greedy customer.

**Comment by ShyamalSankar** [comment]
You could consider including it as a property or creating a new class altogether, depending on your implementation

**Comment by brennanleez-coder** [comment]
> Yes you'll have to use other ways since instanceof is not allowed. For me, I created a method in GreedyCustomer and Customer which returns a String. I then use Customer.method_name().equals("Greedy") to check if it's a greedy customer.

Hi gableejh, i believe that design is a big part in this project and instanceof isnt allowed as it violates the principles. That being said, i believe the customer method you're referring to violates the same principles and will be marked down.

**Comment by gableejh** [comment]
> > Yes you'll have to use other ways since instanceof is not allowed. For me, I created a method in GreedyCustomer and Customer which returns a String. I then use Customer.method_name().equals("Greedy") to check if it's a greedy customer.
> 
> Hi gableejh, i believe that design is a big part in this project and instanceof isnt allowed as it violates the principles. That being said, i believe the customer method you're referring to violates the same principles and will be marked down.

Thanks for sharing brennanleez-coder, didn't thought of that!

**Comment by OuaisBien** [comment]
For me I created a `isGreedy()` for `Customer` which returns false. In the `GreedyCustomer` subclass, it is overridden and returns `true`.

**Comment by ashleyleerx** [comment]
I added a boolean attribute to the Customer class which is set as false.   This boolean attribute will be true for the GreedyCustomer subclass

**Comment by eltonngkk** [comment]
Hi Karin, similar to what others have mentioned, I believe it is fine to set a boolean attribute isGreedy() to check whether it is true or false. Additionally, this checking should be done inside your method for getting an available server and not inside the Arrive class. Hope this helps!

**Comment by brigittesantoso** [comment]
Hi Karin,

I created a boolean attribute isGreedy() under Customer class and getter method getIsGreedy(). getIsGreedy() is called under my ArriveEvent to check if customer is greedy before assigning the customer to a server if there are available waiting spots.

Hope this helps!

**Comment by dumbasssoul** [comment]
> Hi Karin, similar to what others have mentioned, I believe it is fine to set a boolean attribute isGreedy() to check whether it is true or false. Additionally, this checking should be done inside your method for getting an available server and not inside the Arrive class. Hope this helps!

hello where should the getting an available server method be? as i implemented it inside my arrive class but im not sure whether its wrong

**Comment by ydymegan** [comment]
> > Hi Karin, similar to what others have mentioned, I believe it is fine to set a boolean attribute isGreedy() to check whether it is true or false. Additionally, this checking should be done inside your method for getting an available server and not inside the Arrive class. Hope this helps!
> 
> hello where should the getting an available server method be? as i implemented it inside my arrive class but im not sure whether its wrong

Hi, my getting / checking an available server method is also in my ArriveEvent class and it works fine so yours should be alright! However, within the method, I separate to 2 different methods of checking for Greedy and Normal customers.

**Comment by sebin0817** [comment]
> > Hi Karin, similar to what others have mentioned, I believe it is fine to set a boolean attribute isGreedy() to check whether it is true or false. Additionally, this checking should be done inside your method for getting an available server and not inside the Arrive class. Hope this helps!
> 
> hello where should the getting an available server method be? as i implemented it inside my arrive class but im not sure whether its wrong

Hi not sure if you solved it yet, but the following method is embedded on my Arrive Event class! i implemented a method that calls the next event (for arrivalEvent, it returns Serve, Wait or Leave events depending on the conditions you set).  The method for available server came under one of the conditions for Wait Event for me! all the best
